### PR TITLE
ref(js): Add usePolling to replace very old PollingMixin

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/hooks/usePolling.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const DEFAULT_DELAY = 3000;
+
+/**
+ * How long between polls if the most recent poll requet failed
+ */
+const BACKOFF_DELAY = 10000;
+
+function usePolling({url, handleRecieveData, pollingActive = true}) {
+  /**
+   * Returns true if request was successfull, false otherwise.
```
Comment (malwilley) [inline static/hooks/usePolling.jsx:17]: *successful

File: static/hooks/usePolling.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  /**
+   * Cancels pending polling timeouts
+   */
+  const cancelPolling = useCallback(() => {
+    window.clearTimeout(pollTimeoutRef.current);
+    pollTimeoutRef.current = undefined;
+  }, []);
+
+  /**
+   * Beings the polling cycle
```
Comment (malwilley) [inline static/hooks/usePolling.jsx:41]: Begins?

File: static/hooks/usePolling.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import api from 'app/api';
+
+/**
+ * How long between polls
+ */
+const DEFAULT_DELAY = 3000;
+
+/**
+ * How long between polls if the most recent poll requet failed
```
Comment (malwilley) [inline static/hooks/usePolling.jsx:11]: requet

File: static/hooks/usePolling.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * Beings the polling cycle
+   */
+  const triggerPolling = useCallback(async () => {
+    if (!pollingActive) {
+      return cancelPolling;
+    }
+
+    const wasSuuccess = await makeRequest();
```
Comment (malwilley) [inline static/hooks/usePolling.jsx:48]: Spelling is hard 🙃

**Review by malwilley — APPROVED** [review]

File: static/hooks/usePolling.jsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/hooks/usePolling.jsx:41]: old comment

**Pull request merged**